### DISCUSSION TITLE
kubevela vs DAPR?

### DISCUSSION THREAD

**Discussion by nathanawmk** [kubevela/kubevela#1821] [General]
I have my app deployed across multiple cloud and I want cross-cutting concerns to managed. OAM Kubevela vs DAPR - which one should I pick?

**Comment by wonderflow** [comment]
OAM/KubeVela works in the layer of Application Delivery and Day-2 Operations.
Dapr works in the layer of Application Development runtime, using dapr helps you to integrate with any midderware.

They works in different layers, you can use both KubeVela and dapr.
